metricbeat/module/kubernetes/state_container/state_container_test.go

```diff
@@ -20,14 +20,37 @@
 package state_container
 
 import (
+	"encoding/json"
+	"flag"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"sort"
 	"testing"
 
-	"github.com/elastic/beats/metricbeat/helper/prometheus/ptest"
+	"github.com/mitchellh/hashstructure"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/elastic/beats/libbeat/common"
+
+	mbtest "github.com/elastic/beats/metricbeat/mb/testing"
 )
 
+// TestCases holds the list of test cases to test a metricset
+type TestCases []struct {
+	// MetricsFile containing Prometheus outputted metrics
+	MetricsFile string
+
+	// ExpectedFile containing resulting documents
+	ExpectedFile string
+}
+
+var expectedFlag = flag.Bool("update_expected", false, "Update prometheus expected files")
+
 func TestEventMapping(t *testing.T) {
-	ptest.TestMetricSetEventsFetcher(t, "kubernetes", "state_container",
-		ptest.TestCases{
+	testMetricSetEventsFetcher(t, "kubernetes", "state_container",
```

Comment:
There seems to be already an implementation for reporter V2 (`ptest.TestMetricSet`), could you use it here instead of duplicating it?

If you need to add another implementation I think it'd be better to do it in `ptest` so other metricsets can also take advantage of it.

Reply:
This is exactly what I needed here. Thank you Jaime!